### PR TITLE
feat: default allowed domains for huggingface inference deployments

### LINKED ISSUE #188: [HuggingFace] Add HF domains to allowed domains by default
### Name and Version

ai-dial-admin-deployment-manager-backend [0.14.0]

### What is the problem this feature will solve?

Currently, user always needs to explicitly add HuggingFace domains to allowed domains (firewall settings) to successfully run models.

### What is the feature you are proposing to solve the problem?

Add HuggingFace domains to HF deployment's allowed domains by default.

### What alternatives have you considered?

_No response_

### PR DESCRIPTION
**Pull request by Pasichniuk**
### Applicable issues

- fixes #188 

### Description of changes

Added new property 'app.huggingface.default-allowed-domains' to set domains which will be added to Cilium network policy egress by default for inference deployments with huggingface source.

### Checklist

- [x] Title of the pull request follows [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by ai-dial-actions** [comment]
<h1>Dependency Review</h1>
The following issues were found:<ul><li>✅ 0 vulnerable package(s)</li><li>✅ 0 package(s) with incompatible licenses</li><li>✅ 0 package(s) with invalid SPDX license definitions</li><li>⚠️ 2 package(s) with unknown licenses.</li></ul>
See the Details below.<h2>License Issues</h2>
<h4><em>settings.gradle</em></h4>
<table><tr><td>Package</td><td>Version</td><td>License</td><td>Issue Type</td></tr><tr><td>com.gradle:common-custom-user-data-gradle-plugin</td><td>2.1</td><td>Null</td><td>Unknown License</td></tr><tr><td>com.gradle:develocity-gradle-plugin</td><td>4.2</td><td>Null</td><td>Unknown License</td></tr></table>
<h2>OpenSSF Scorecard</h2>
<table><tr><th>Package</th><th>Version</th><th>Score</th><th>Details</th></tr>
<tr><td> maven/com.gradle:common-custom-user-data-gradle-plugin </td><td>2.1</td>
      <td> Unknown</td><td>Unknown</td></tr>
<tr><td> maven/com.gradle:develocity-gradle-plugin </td><td>4.2</td>
      <td> Unknown</td><td>Unknown</td></tr>
</table><h2>Scanned Files</h2>
<ul><li>settings.gradle</li></ul>

File: src/test/java/com/epam/aidial/deployment/manager/service/deployment/InferenceDeploymentManagerTest.java [hunk 1]
```diff
@@ -422,6 +426,64 @@ void deploy_shouldDeployInferenceService() {
...
+                inferenceManifestGenerator,
+                containerPortResolver,
+                ciliumNetworkPolicyCreator,
+                deploymentRepository,
+                k8sKserveClient,
+                kserveDeployProperties, huggingFaceProperties
+        );
+    }
+
     @Test
```
Comment (oleksii-donets) [inline src/test/java/com/epam/aidial/deployment/manager/service/deployment/InferenceDeploymentManagerTest.java:575]: Consider adding edge-case tests:
- **Non-HuggingFace source** — verify that default domains are NOT appended when the source is not `InferenceDeploymentHuggingFaceSource`
- **Null/empty deployment `allowedDomains` with HF source** — only defaults should appear
- **Empty `defaultAllowedDomains` with HF source** — only deployment domains should appear

File: src/main/java/com/epam/aidial/deployment/manager/service/deployment/AbstractDeploymentManager.java [hunk 2]
```diff
@@ -785,10 +785,15 @@ private void deleteCiliumNetworkPolicy(String name) {
     protected Set<Integer> getCiliumIngressPorts(D deployment) {
         Set<Integer> ports = new HashSet<>();
         Optional.ofNullable(deployment.getContainerPort()).ifPresent(ports::add);
-        Optional.ofNullable(defaultContainerPort).ifPresent(ports::add);
+        Optional.of(defaultContainerPort).ifPresent(ports::add);
```
Comment (oleksii-donets) [inline src/main/java/com/epam/aidial/deployment/manager/service/deployment/AbstractDeploymentManager.java]: This change from `Optional.ofNullable` to `Optional.of` is unrelated to the feature. Since `defaultContainerPort` is a primitive `int`, the `Optional` wrapping is unnecessary altogether — this could just be:
```java
ports.add(defaultContainerPort);
```
Suggest either simplifying it fully or reverting this change and addressing it in a separate cleanup PR.

File: src/main/java/com/epam/aidial/deployment/manager/service/deployment/InferenceDeploymentManager.java [hunk 3]
```diff
@@ -50,14 +56,16 @@ public InferenceDeploymentManager(
...
+            KserveDeployProperties kserveDeployProperties,
+            HuggingFaceProperties huggingFaceProperties
     ) {
         super(k8sClient, disposableResourceManager, manifestGenerator, deploymentRepository,
                 containerPortResolver, ciliumNetworkPolicyCreator, kserveDeployProperties.getNamespace(),
                 kserveDeployProperties.getStartupTimeout(), DEFAULT_KSERVE_SERVICE_PORT);
         this.inferenceManifestGenerator = inferenceManifestGenerator;
         this.k8sKserveClient = k8sKserveClient;
         this.useClusterInternalUrl = kserveDeployProperties.isUseClusterInternalUrl();
+        this.defaultAllowedDomains = huggingFaceProperties.getDefaultAllowedDomains();
```
Comment (oleksii-donets) [inline src/main/java/com/epam/aidial/deployment/manager/service/deployment/InferenceDeploymentManager.java:67]: The list returned by `getDefaultAllowedDomains()` is mutable (`Collectors.toList()`). Consider wrapping with `List.copyOf()` to prevent accidental mutation:
```java
this.defaultAllowedDomains = List.copyOf(huggingFaceProperties.getDefaultAllowedDomains());
```

File: src/main/java/com/epam/aidial/deployment/manager/service/deployment/InferenceDeploymentManager.java [hunk 4]
```diff
@@ -228,6 +236,22 @@ protected String getServiceNameLabel() {
...
+        List<String> allowedDomains = super.getEffectiveDeploymentAllowedDomains(deployment);
+
+        if (!(deployment.getSource() instanceof InferenceDeploymentHuggingFaceSource)
+                || CollectionUtils.isEmpty(defaultAllowedDomains)) {
+            return allowedDomains;
+        }
+
+        return Stream.concat(
+                        allowedDomains.stream(),
+                        defaultAllowedDomains.stream().filter(StringUtils::isNotBlank))
```
Comment (oleksii-donets) [inline src/main/java/com/epam/aidial/deployment/manager/service/deployment/InferenceDeploymentManager.java]: nit: `.filter(StringUtils::isNotBlank)` is redundant here — `getDefaultAllowedDomains()` already filters out blank/empty strings during parsing.

File: src/main/java/com/epam/aidial/deployment/manager/huggingface/properties/HuggingFaceProperties.java [hunk 5]
```diff
@@ -1,18 +1,42 @@
...
+
+    @Getter(AccessLevel.NONE)
+    private String defaultAllowedDomains;
+
+    /**
+     * Returns the default allowed domains for HuggingFace inference deployments egress.
+     * Parses the comma-separated string {@link #defaultAllowedDomains}.
+     */
+    public List<String> getDefaultAllowedDomains() {
+        if (StringUtils.isBlank(defaultAllowedDomains)) {
```
Comment (oleksii-donets) [inline src/main/java/com/epam/aidial/deployment/manager/huggingface/properties/HuggingFaceProperties.java]: This re-parses the comma-separated string on every call. Currently mitigated since `InferenceDeploymentManager` caches the result in its constructor, but future callers could hit this repeatedly. Consider parsing once (e.g., via `@PostConstruct` or lazy init) and returning an unmodifiable list.

**Review by oleksii-donets — COMMENTED** [review]
Overall a well-structured PR with a clean extensible design pattern. Left a few suggestions around test coverage, immutability, and a minor unrelated change.

File: src/test/java/com/epam/aidial/deployment/manager/service/deployment/InferenceDeploymentManagerTest.java (hunk 1 above)
Comment (Pasichniuk) [reply src/test/java/com/epam/aidial/deployment/manager/service/deployment/InferenceDeploymentManagerTest.java:575]: Added tests

**Comment by ai-dial-actions** [comment]
<h1>Dependency Review</h1>
The following issues were found:<ul><li>✅ 0 vulnerable package(s)</li><li>✅ 0 package(s) with incompatible licenses</li><li>✅ 0 package(s) with invalid SPDX license definitions</li><li>⚠️ 2 package(s) with unknown licenses.</li></ul>
See the Details below.<h2>License Issues</h2>
<h4><em>settings.gradle</em></h4>
<table><tr><td>Package</td><td>Version</td><td>License</td><td>Issue Type</td></tr><tr><td>com.gradle:common-custom-user-data-gradle-plugin</td><td>2.1</td><td>Null</td><td>Unknown License</td></tr><tr><td>com.gradle:develocity-gradle-plugin</td><td>4.2</td><td>Null</td><td>Unknown License</td></tr></table>
<h2>OpenSSF Scorecard</h2>
<table><tr><th>Package</th><th>Version</th><th>Score</th><th>Details</th></tr>
<tr><td> maven/com.gradle:common-custom-user-data-gradle-plugin </td><td>2.1</td>
      <td> Unknown</td><td>Unknown</td></tr>
<tr><td> maven/com.gradle:develocity-gradle-plugin </td><td>4.2</td>
      <td> Unknown</td><td>Unknown</td></tr>
</table><h2>Scanned Files</h2>
<ul><li>settings.gradle</li></ul>

File: src/main/java/com/epam/aidial/deployment/manager/service/deployment/AbstractDeploymentManager.java (hunk 2 above)
Comment (Pasichniuk) [reply src/main/java/com/epam/aidial/deployment/manager/service/deployment/AbstractDeploymentManager.java]: Removed Optional usage for port completely

File: src/main/java/com/epam/aidial/deployment/manager/service/deployment/InferenceDeploymentManager.java (hunk 3 above)
Comment (Pasichniuk) [reply src/main/java/com/epam/aidial/deployment/manager/service/deployment/InferenceDeploymentManager.java:67]: Refactored getter to return immutable list

File: src/main/java/com/epam/aidial/deployment/manager/service/deployment/InferenceDeploymentManager.java (hunk 4 above)
Comment (Pasichniuk) [reply src/main/java/com/epam/aidial/deployment/manager/service/deployment/InferenceDeploymentManager.java]: Removed

File: src/main/java/com/epam/aidial/deployment/manager/huggingface/properties/HuggingFaceProperties.java (hunk 5 above)
Comment (Pasichniuk) [reply src/main/java/com/epam/aidial/deployment/manager/huggingface/properties/HuggingFaceProperties.java]: Done, thanks

**Review by oleksii-donets — APPROVED** [review]
